impress-org/givewp
# Refactor: replace donation form block modal

Resolves [GIVE-267] & [GIVE-385]

## Description
This Pull Request introduces improvements to the DonationForm block by replacing the existing form modal with Modal components provided by react-aria. By implementing a more standardized Modal component, we ensure consistency between the form preview in the editor and the frontend view.:

Our forms lack a width and are fluid, causing issues with the react iframe resizer's ability to reliably obtain and resize. To ensure the form is fully displayed while in the modal view the dialog must maintain 100% width. 

## Related changes
Styles applied in the previous modals stylesheet have been removed that had conflicts with the WP block editors modal display. These conflicts created visual issyes while renaming a block & for the preferences menu.

## Affects
Donation Form Block

## Visuals

https://github.com/impress-org/givewp/assets/75056371/6d5c1b59-f3a4-4313-bbf2-94feeee32030


<img width="2051" alt="Screen Shot 2024-03-12 at 7 57 14 AM" src="https://github.com/impress-org/givewp/assets/75056371/12e2ac45-8be1-4b45-9631-be25c723dd1b">
<img width="2051" alt="Screen Shot 2024-03-12 at 7 57 24 AM" src="https://github.com/impress-org/givewp/assets/75056371/84429a1b-27c6-43ef-a8cd-7a003d55bff6">

## Testing Instructions

- Add a DonationFormBlock for the varius form templates
- Set the view setting for the block to Modal
- Verify the the modal is scrollable
- Verify the modal works in the editor & on the frontend

- Go to the WP block editor -> add a row and rename the block. Verify no visual issues with the modal.
- Select the 3-dots on the top right to open the settings menu -> select preferences. Verify no visual issues with the preferences modal.
## Pre-review Checklist




[GIVE-267]: https://stellarwp.atlassian.net/browse/GIVE-267?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GIVE-385]: https://stellarwp.atlassian.net/browse/GIVE-385?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
